### DISCUSSION TITLE
Uncertainty analysis with Monte Carlo

### DISCUSSION THREAD

**Discussion by LFGS20** [gempy-project/gempy#972] [Q&A] [answered]
Hello, I want to perform an uncertainty analysis with the Monte Carlo method to my geological model of gempy made with version 2024.2.0.2, I was looking at some examples with previous versions and I see that they use the attribute geo_model.surfaces to access the units/layers of the model, but for my model I have not been able to access this information of the units because this new version does not handle this attribute,   I have only been able to access the information of the points of the surfaces with geo_model.surface_points_copy and with this information I cannot perform the uncertainty analysis.
Is there any way or example where I can see how this analysis is performed with the 2024.2.0.2 version of gempy?

**Comment by NilsChudalla** [comment, marked as answer]
Dear @LFGS20 ,
If you are looking for vertices of your meshes, this is how you can find them in the current version of gempy:
<code>geo_model.solutions.dc_meshes[i].vertices</code>, where i is the index of the surface from your stack. Note, that the coordiantes are currently transformed to optimize the computation. You can easily retransform the vertices to real life coordinates by
<code>geo_model.input_transform.apply_inverse(geo_model.solutions.dc_meshes[0].vertices)</code>

if you are looking for the lithblock solution (essentially the voxels of your model), you might want to go with
<code>geo_model.solutions.raw_arrays.lith_block</code>

To come up with an example, how the uncertainty analysis can be performed, can you be a bit more specific about what the aim of your analysis is? Are you looking into spatial uncertainty of your input or parameter uncertainty (e.g. density in a gravity context)? Are you only doing a monte carlo simulation, where you perturb model input, or are you aiming for Metropolis Hastings or Hamiltonian Methods, where you want to sample from the posterior? Also, an example notebook is always helpful

**Reply by LFGS20** [reply]
Hi, the objective of my analysis is the spatial uncertainty of my input, performing only a Monte Carlo simulation, where the value of z is perturbed, to obtain the probability of occurrence of my layers.

**Reply by NilsChudalla** [reply]
Okay, so most likely you want to have the lith_block then. It comes out as a 1D array, so make sure to reshape it according to your model extent :) Does this resolve your issue?

**Reply by NilsChudalla** [reply]
@LFGS20 does this solve your question? If so, I would close the topic.

**Reply by LFGS20** [reply]
Yes, thank you

**Reply by NilsChudalla** [reply]
And also here, very welcome :) If you feel like this point needs further discussion, feel free to reopen or start a new thread :)
